### DISCUSSION TITLE
Getting error in chapter 4, DataLoader error.

### DISCUSSION THREAD

**Discussion by hiteshmaurya99** [mrdbourke/pytorch-deep-learning#141] [Q&A] [answered]
Here is my code.
from torch.utils.data import DataLoader

train_dataloader = DataLoader(dataset = train_data,
                        batch_size = 1,
                        shuffle = True
                        num_workers = 1)
test_dataloader = DataLoader(dataset = test_data,
                       batch_size = 1,
                       shuffle = False,
                       num_workers = 1)
train_dataloader, test_dataloader
##This works exactly as the vedio.

img, label = next(iter(train_dataloader))
img, label
##This part is the problem and raises error:

TypeError: Caught TypeError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/torch/utils/data/_utils/worker.py", line 302, in _worker_loop
    data = fetcher.fetch(index)
  File "/usr/local/lib/python3.7/dist-packages/torch/utils/data/_utils/fetch.py", line 49, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/usr/local/lib/python3.7/dist-packages/torch/utils/data/_utils/fetch.py", line 49, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
TypeError: 'DataLoader' object is not subscriptable

**Comment by Samer98** [comment, marked as answer]
look i had the same error "TypeError: Caught TypeError in DataLoader worker process 0." but somehow was the issue in the  data_transform
because i used trivailaagumanted(magnitude_bins=1). Value 1 had the issue

**Reply by hiteshmaurya99** [reply]
Thank you friend.
